### PR TITLE
added pundit

### PR DESCRIPTION
**Pull request by alanhala**
### Summary
- included pundit gem in the Gemfile
- created the article policy
- added the authorize method in the controller and the redirect in case of unauthorized exception 

### Trello Card
https://trello.com/c/51BV3wHh/16-instalar-y-configurar-pundit

### CODE REVIEW

File: app/policies/application_policy.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
```
Comment (mdesanti) [inline app/policies/application_policy.rb]: Why is this method returning false?

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    false
+  end
+
+  def show?
+    scope.where(:id => record.id).exists?
+  end
+
+  def create?
```
Comment (mdesanti) [inline app/policies/application_policy.rb]: Why is this method returning false?

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    false
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
```
Comment (mdesanti) [inline app/policies/application_policy.rb]: Why is this method returning false?

File: app/policies/application_policy.rb (hunk 1 above)
Comment (alanhala) [inline app/policies/application_policy.rb]: @mdesanti It is the default policy. I should remove it.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    flash[:error] = "You are not authorized to perform this action."
+    redirect_to @article
```
Comment (mdesanti) [inline app/controllers/application_controller.rb]: Be careful with this. You will not always have the @article. Maybe redirect to root?

Comment (alanhala) [inline app/controllers/application_controller.rb]: @mdesanti maybe asking if the @article exists, and if not redirect to root. What do you think?

Comment (mdesanti) [inline app/controllers/application_controller.rb]: The thing is that if the app grows, you might have multiple entities. You will not be asking for each of the possible models to see if the exist. Besides, the exception might be raised when trying to access the `new` method, and when you check permissions, you don't have an @article.

**Comment by mdesanti** [comment]
:+1:

**Pull request merged**